ocrd/ocrd/workspace.py

```diff
@@ -276,12 +276,18 @@ def rename_file_group(self, old, new):
             url_replacements = {}
             log.info("Moving files")
             for mets_file in self.mets.find_files(fileGrp=old, local_only=True):
-                new_url = sub(r'^%s/' % old, '%s/' % new, mets_file.url)
+                new_url = mets_file.url
+                # Directory part
+                new_url = sub(r'^%s/' % old, r'%s/' % new, new_url)
+                # File part
+                new_url = sub(r'/%s' % old, r'/%s' % new, new_url)
                 url_replacements[mets_file.url] = new_url
                 # move file from ``old`` to ``new``
                 move(mets_file.url, new_url)
                 # change the url of ``mets:file``
                 mets_file.url = new_url
+                # change the file ID and update structMap
+                mets_file.ID = sub(r'^%s' % old, r'%s' % new, mets_file.ID)
```

Comment:
Just a precaution: maybe we should check whether `mets_file.mets` already contains the target ID – and then keep the old one. (For example, someone might have used `rename-group` earlier, and now tries to rename back with the new implementation...)

Reply:
You mean something like this?

```python
# change the file ID and update structMap                                                                    
new_id = sub(r'^%s' % old, r'%s' % new, mets_file.ID)                                                        
try:                                                                                                         
    next(self.mets.find_files(ID=new_id))                                                                    
    log.warning("ID %s already exists, not changing ID while renaming %s -> %s" % (new_id, old_url, new_url))
except StopIteration:
    mets_file.ID = new_id                                                                                    
```

Reply:
yes, that's perfect!

Reply:
5158e152